x-pack/test/security_solution_api_integration/test_suites/investigation/saved_objects/trial_license_complete_tier/notes.ts

```diff
@@ -70,5 +80,428 @@ export default function ({ getService }: FtrProviderContext) {
+        expect(notes.length).to.be(1);
+        expect(notes[0].note).to.be('second note');
+      });
+
+      it('should retrieve considering search query parameter', async () => {
+        const eventId = uuidv4();
+        const timelineId = uuidv4();
+
+        await createNoteAssociatedWithDocumentOnly(
+          supertest,
+          eventId,
+          'note associated with an event only'
+        );
+        await createNoteAssociatedWithSavedObjectOnly(
+          supertest,
+          timelineId,
+          'note associated with a timeline only'
+        );
+        await createNoteAssociatedWithDocumentAndSavedObject(
+          supertest,
+          eventId,
+          timelineId,
+          'note associated with an event and a timeline'
+        );
+        await createNoteAssociatedWithNothing(supertest, 'note associated with nothing');
+
+        const response = await supertest
+          .get('/api/note?search=event')
+          .set('kbn-xsrf', 'true')
+          .set('elastic-api-version', '2023-10-31');
+
+        const { totalCount } = response.body;
+
+        expect(totalCount).to.be(2);
+      });
+
+      // TODO why can't we sort on every field? (I tested for the note field (or a random field like abc) and the endpoint crashes)
+      it('should retrieve considering sortField query parameters', async () => {
+        await createNoteAssociatedWithDocumentOnly(supertest, '1', 'note 3');
+        await createNoteAssociatedWithDocumentOnly(supertest, '2', 'note 2');
+        await createNoteAssociatedWithDocumentOnly(supertest, '3', 'note 1');
+
+        const response = await supertest
+          .get('/api/note?sortField=eventId')
+          .set('kbn-xsrf', 'true')
+          .set('elastic-api-version', '2023-10-31');
+
+        const { totalCount, notes } = response.body;
+
+        expect(totalCount).to.be(3);
+        expect(notes[0].eventId).to.be('1');
+        expect(notes[1].eventId).to.be('2');
+        expect(notes[2].eventId).to.be('3');
+      });
+
+      it('should retrieve considering sortOrder query parameters', async () => {
+        await createNoteAssociatedWithDocumentOnly(supertest, '1', 'note 3');
+        await createNoteAssociatedWithDocumentOnly(supertest, '2', 'note 2');
+        await createNoteAssociatedWithDocumentOnly(supertest, '3', 'note 1');
+
+        const response = await supertest
+          .get('/api/note?sortField=eventId&sortOrder=desc')
+          .set('kbn-xsrf', 'true')
+          .set('elastic-api-version', '2023-10-31');
+
+        const { totalCount, notes } = response.body;
+
+        expect(totalCount).to.be(3);
+        expect(notes[0].eventId).to.be('3');
+        expect(notes[1].eventId).to.be('2');
+        expect(notes[2].eventId).to.be('1');
+      });
+
+      // TODO should add more tests for the filter query parameter (I don't know how it's supposed to work)
+
+      // TODO should add more tests for the MAX_UNASSOCIATED_NOTES advanced settings values
+    });
```

Comment:
Do you still want to do this before this PR is merged?

Reply:
no, there are some unknowns on my side, I just added these todos when we revisit the api, probably around 9.0